src/main/java/it/gov/pagopa/merchant/service/RetrieveMerchantIdServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package it.gov.pagopa.merchant.service;
+
+import it.gov.pagopa.merchant.constants.MerchantConstants;
+import it.gov.pagopa.merchant.dto.MerchantInfoDTO;
+import it.gov.pagopa.merchant.exception.ClientExceptionWithBody;
+import it.gov.pagopa.merchant.repository.MerchantRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RetrieveMerchantIdServiceImpl implements RetrieveMerchantIdService{
```

Comment:
I'd use a more generic name such as MerchantInfoService. Furthermore, verbs are generally used to name methods rather than classes or attributes [follows from above]